src/AppInstallerRepositoryCore/Rest/Schema/1_0/Interface.cpp

```diff
@@ -0,0 +1,202 @@
+            std::optional<std::string> packageName = GetStringFromJsonStringValue(manifestItem.at(GetJsonKeyNameString(PackageName)));
+            std::optional<std::string> publisher = GetStringFromJsonStringValue(manifestItem.at(GetJsonKeyNameString(Publisher)));
+            std::optional<std::string> packageFamilyName = GetStringFromJsonStringValue(manifestItem.at(GetJsonKeyNameString(PackageFamilyName)));
+            std::optional<std::string> productCode = GetStringFromJsonStringValue(manifestItem.at(GetJsonKeyNameString(ProductCode)));
+            web::json::value versionValue = manifestItem.at(GetJsonKeyNameString(Versions));
+
+            if (!packageId.has_value() || !packageName.has_value() || !publisher.has_value() || versionValue.is_null() || versionValue.as_array().size() == 0)
+            {
+                AICLI_LOG(Repo, Verbose, << "Received incomplete package. Skipping package: " << packageId.value_or(""));
+                continue;
+            }
+
+            std::vector<AppInstaller::Utility::VersionAndChannel> versionList;
+            for (auto& versionItem : versionValue.as_array())
+            {
+                std::optional<std::string> version = GetStringFromJsonStringValue(versionItem.at(GetJsonKeyNameString(Version)));
+                std::optional<std::string> channel = GetStringFromJsonStringValue(versionItem.at(GetJsonKeyNameString(Channel)));
+
+                if (!version.has_value())
+                {
+                    AICLI_LOG(Repo, Verbose, << "Received incomplete package version. Skipping version from package: " << packageId.value());
+                    continue;
+                }
+
+                versionList.emplace_back(AppInstaller::Utility::VersionAndChannel(std::move(version.value()), std::move(channel.value_or(""))));
+            }
+
+            if (versionList.size() == 0)
+            {
+                AICLI_LOG(Repo, Verbose, << "Received no valid versions. Skipping package: " << packageId.value());
+                continue;
+            }
+
+            PackageInfo packageInfo = PackageInfo(std::move(packageId.value()), std::move(packageName.value()), std::move(publisher.value()));
+            Package package = Package(std::move(packageInfo), std::move(versionList));
+            result.Matches.emplace_back(std::move(package));
+        }
+
+        return result;
+    }
+
+    std::optional<Manifest::Manifest> Interface::GetManifestByVersion(const std::string& packageId, const std::string& version, const std::string& channel) const
+    {
+        HttpClientHelper clientHelper(GetManifestByVersionEndpoint(m_restApiUri, packageId, version, channel));
+        web::json::value jsonObject = clientHelper.HandleGet();
+
+        if (jsonObject.is_null())
+        {
+            return {};
+        }
+
+        // Parse json and return Manifest
+        auto& manifestObject = jsonObject.at(GetJsonKeyNameString(Data));
+        UNREFERENCED_PARAMETER(manifestObject);

```

Comment:
nit: you can just (void)jsonObject.at(GetJsonKeyNameString(Data)); So you don't need the UNREFERENCED_PARAMETER below